cpp_linter/run.py

```diff
@@ -115,17 +121,18 @@
 cli_arg_parser.add_argument(
     "-l",
     "--lines-changed-only",
-    default="false",
-    type=lambda input: input.lower() == "true",
-    help="Set this option to 'true' to only analyse changes in the event's diff. "
+    default=0,
+    type=lambda a: 1 if a.lower() == "true" else (2 if a.lower() == "strict" else 0),
+    help="Set this option to 'true' to only analyze changes in the event's diff. "
+    "Set this to 'strict' to only analyze additions in the event's diff. "
     "Defaults to %(default)s.",
 )
 cli_arg_parser.add_argument(
     "-f",
     "--files-changed-only",
     default="false",
     type=lambda input: input.lower() == "true",
-    help="Set this option to 'false' to analyse any source files in the repo. "
+    help="Set this option to 'false' to analyze any source files in the repo. "
     "Defaults to %(default)s.",
```

Comment:
`files-changed-only` argument default value here is false but on README.md, it is noted 'ture'

Reply:
This is so I can run the action locally without a docker. Using this feature currently requires a REST API token, and I haven't regenerated mine in a while. I suppose we could use git directly, but I'm not sure how to identify a commit that is in an open PR. So, if we switch to `git show`, we could only support push events on local machines.

Reply:
Remember that all defaults from actions.yml are passed to the `cpp-linter` executable script when on a github runner. Unless users are running the action as a pkg in a github runner, then this may cause undesired results. The google/flatbuffers project was using the action.yml defaults.

Reply:
OK, I see

Reply:
> This is so I can run the action locally without a docker. 

This is very powerful. If possible I recommend users to use it as an action than a package, more users who used the used by section can display more count, then will attract more users to use it, the only drawback is it takes a little time than run as a package.

Reply:
I'm trying to pivot away from the docker since it seems to have trouble reading the database and third party dependencies or using cross-compilers. See [github docs about creating a composite action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action). This will allow us to use clang-tools-pip + cpp-linter-action without a docker... But we're not quite there yet.

Reply:
1. I'd like to start releasing this pkg to pypi first, so we can install the distributed wheel (faster than `pip install .`).
2. I'd also like to start hosting our own releases of clang-tools-static-binaries to ensure reliability.

We're going to have to make the transition to composite action in phases.

Reply:
I thinking about whether to create a new repository called `cpp-linter` which is the core repo of Org `cpp-linter`. Then move out the cpp_linter source from `cpp-linter-action` repository, only keep the necessary action part code, it will the action repo very simple.

The core repo supports GitHub API, but we are open to supporting GitLab, and Bitbucket if need to have it in the future,

Reply:
I agree with the two points you mentioned above. 
Again, it should be more good to release pkg to PyPI from the new repo `cpp-linter`

Reply:
Yes abstracting the python source into a separate repo seems appropriate.